crates/hc_sandbox/src/bundles.rs

```diff
@@ -0,0 +1,123 @@
+//! Helpers for working with dna files.
+use std::path::Path;
+use std::path::PathBuf;
+
+use anyhow::bail;
+use anyhow::ensure;
+use walkdir::WalkDir;
+
+#[derive(Debug, Clone)]
+/// Either a set of Dnas or a hApp bundle.
+pub enum DnasHapp {
+    /// Set of Dnas to install.
+    Dnas(Vec<PathBuf>),
+    /// Happ bundle to install.
+    HApp(Option<PathBuf>),
+}
+
+impl DnasHapp {
+    /// Create a new DnasHapp from either dnas or happ bundle.
+    pub fn new(dnas: Option<Vec<PathBuf>>, happ: Option<Option<PathBuf>>) -> Self {
+        match (dnas, happ) {
+            (Some(dnas), None) => Self::Dnas(dnas),
+            (None, Some(happ)) => Self::HApp(happ),
+            _ => unreachable!("Cannot have both dnas and happ"),
```

Comment:
If this type is being used, and the arguments come straight from the commandline, is this definitely unreachable? I guess only if using structopt "mutually exclusive" args or whatever.